go-minitwit/db_methods.go

```diff
@@ -230,76 +218,129 @@ func getUserByUsername(userName string) ([]map[string]interface{}, error) {
 
 // registers a new user
 func registerUser(userName string, email string, password [16]byte) error {
-	query := `insert into user (username, email, pw_hash) values (?, ?, ?)`
-	var db, err = connect_db(DATABASE)
-	if err != nil {
-		return err
+
+	pwHashString := hex.EncodeToString(password[:])
+
+	newUser := User{
+		Username: userName,
+		Email:    email,
+		PwHash:   pwHashString,
 	}
-	args := []interface{}{userName, email, pq.Array(password)}
-	messages, err := query_db(db, query, args, false)
-	fmt.Println("this is the messages", messages)
-	return err
+
+	dbNew.Create(&newUser)
+
+	if dbNew.Error != nil {
+		log.Fatal(dbNew.Error)
+		return dbNew.Error
+	}
+
+	return nil
+
 }
 
 // adds a new message to the database
 func addMessage(text string, author_id string) error {
-	query := `insert into message (author_id, text, pub_date, flagged) values (?, ?, ?, 0)`
-	var db, err = connect_db(DATABASE)
+
+	authorIDintValue, err := strconv.Atoi(author_id)
```

Comment:
Here I would change author_id to int since we are converting to String to use the function and then converting back again to String. I think we should review all the usage of strconv Atoi and Itoa and define structures and functions where we don't need to do it. Probably we will increase a lot the performance. We don't need to do it in this PR but it is something to be tracked as a future task.

Reply:
changing...